core/templates/components/state-editor/state-skill-editor/state-skill-editor.directive.ts

```diff
@@ -0,0 +1,135 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the Skill editor section in the state editor.
+*/
+
+require(
+  'domain/topics_and_skills_dashboard/' +
+  'topics-and-skills-dashboard-backend-api.service.ts');
+require('components/skill-selector/select-skill-modal.controller.ts');
+require(
+  'components/skill-selector/skill-selector.component.ts');
+require('pages/story-editor-page/services/story-editor-state.service.ts');
+require('services/alerts.service.ts');
+require('services/contextual/window-dimensions.service.ts');
+
+angular.module('oppia').directive('stateSkillEditor', [
+  'UrlInterpolationService', function(UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {
+        onSaveLinkedSkillId: '=',
+        onSaveStateContent: '='
+      },
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/components/state-editor/state-skill-editor/' +
+        'state-skill-editor.directive.html'),
+      controller: [
+        '$rootScope', '$scope', '$uibModal', 'AlertsService',
+        'StateSkillService', 'StoryEditorStateService',
+        'TopicsAndSkillsDashboardBackendApiService',
+        'WindowDimensionsService',
+        function(
+            $rootScope, $scope, $uibModal, AlertsService,
+            StateSkillService, StoryEditorStateService,
+            TopicsAndSkillsDashboardBackendApiService,
+            WindowDimensionsService) {
+          var ctrl = this;
+          var categorizedSkills = null;
+          var untriagedSkillSummaries = null;
+          var _init = function() {
+            TopicsAndSkillsDashboardBackendApiService.fetchDashboardDataAsync()
+              .then(function(response) {
+                categorizedSkills = response.categorizedSkillsDict;
+                untriagedSkillSummaries = response.untriagedSkillSummaries;
+                $rootScope.$applyAsync();
+              });
+          };
+
+          $scope.addSkillModalAsync = async function() {
```

Comment:
Should this be openAddSkillModal()?

Also, why does it need to be async? I don't see the difference between this and the delete modal in terms of that.

Reply:
Modified.